free-3.md

```diff
@@ -10,41 +10,104 @@ _What need is met by your idea? Who are you helping? What is that person's
+for this is because with a program it is easy to add a ton of features and then
+dilute the program, confusing the user. A DSL would probably be appropriate because
+they can be abstracted to allow for a non-technical user to easily make music.
+I think a language adresses the need for easier interactions with the music. 
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I may not be big on making music but I am a huge music fan - especially
+for music that has cool beats. I also would be really enjoy plugging something
+like this into speakers and freestyle with it. Additionally this idea excites
+me simply because making a language like this would be really interesting.
+
+I was thinking about a possible music DSL from the start but hadnt created
+a good idea. Then at some point I was playing music really loud and kept
+tapping my finger to the beat. I kinda thought it would be cool if I could
+actually make music by doing that and it kinda sprung up that it could be a
+DSL.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Simple beat creation.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+A user would interact with the language by asking the language to record a sound
+and then clicking on the page to create the sound. Idealy you could alter the pitch
+of the beat by changing where you click. Also I may or may not want the user to be
+able to alter the sound after recording it. Lastly, the program would record where the
+button presses were made to display to the user. Programming in this language would look
+a lot like clicking on a blank screen that displays where you clicked and then makes music
+from it. I think this is the right way to interact with the domain because it is super simple.
+A mouseclick is very intuitive/fluent for everyone who can easily use one and is a fairly
+natural way of interacting with beats since many people already dance or tap to the beat.
+
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+I think when a program runs the user will click on some screen somewhere and the
+program will record the clicks and translate it into sound. It would also display
+where they clicked so they could see the pitch created. I am imagining to make
+things consistent it would be good to record the user input and keep looping over it
+allowing the user to make slight changes to the beat or create a new beat. The
+programs only interaction with the user would probably be to play the music and
+display general HOWTO information. Also the program would allow the user to move
+the "beats" around. In terms of errors though, I cannot imagine any errors occuring.
```

Comment:
Another possible error may be trying to click out of the allowed area because even if it's a full screen, most computers allow you to move the mouse off screen. Would the program leave it, assuming that the user did not mean to use this click? Or would the program put it at the nearest possible point? Of course, users may come up with other ways to "break" the language and how to deal with those errors can be decided later, but errors don't always have to stop the program.